go-runtime/compile/schema.go

```diff
@@ -667,6 +667,9 @@ func visitType(pctx *parseContext, node ast.Node, tnode types.Type) (schema.Type
 			} else if !strings.HasPrefix(nodePath, pctx.pkg.PkgPath) {
 				// If this type is named and declared in another module, it's a reference.
 				// The only basic-typed references supported are enums.
+				if !strings.HasPrefix(named.Obj().Pkg().Path(), "ftl/") {
```

Comment:
This isn't going to interfere with the types in `go-runtime/ftl` like `ftl.Option` is it?

Reply:
Just gave it a try and it seemed to work
```
type EchoResponse struct {
	Message         string                    `json:"message"`
	OptionalString ftl.Option[string]        `json:"optionalString"` // This works fine
	OptionalDID     ftl.Option[did.BearerDID] `json:"optionalDID"` // This fails when included
}
```